mon-pix/app/controllers/authenticated/user-account/language.js

```diff
@@ -3,14 +3,18 @@ import { action } from '@ember/object';
 import { inject as service } from '@ember/service';
 
 export default class UserAccountPersonalInformationController extends Controller {
+  @service currentUser;
+  @service dayjs;
   @service intl;
   @service currentDomain;
-  @service location;
 
   @action
-  async onLanguageChange(value) {
+  async onLanguageChange(language) {
     if (!this.currentDomain.isFranceDomain) {
-      this.location.replace(`/mon-compte/langue?lang=${value}`);
+      this.intl.setLocale([language]);
+      this.dayjs.setLocale(language);
+
+      await this.currentUser.user.save({ adapterOptions: { lang: language } });
```

Comment:
```suggestion
      await this.currentUser.user.save({ adapterOptions: { lang: language } });
      
      this.intl.setLocale([language]);
      this.dayjs.setLocale(language);
```

J'aurai fait plutôt ceci. Si l'appel vers l'API n'est pas _OK_ à ce moment là l'interface ne changera pas vu que l'erreur stoppera le processus.

Reply:
Bonne idée, je fais ça 😃